vm/string_test.go

```diff
@@ -59,18 +73,92 @@ func TestEvalInfixStringExpression(t *testing.T) {
 		{`" ".reverse`, " "},
 		{`"-123".reverse`, "321-"},
 		{`"Hello\nWorld".reverse`, "dlroW\nolleH"},
+		{`"Hello hello HeLlo".delete("el")`, "Hlo hlo HeLlo"},
+		{`"".empty`, TRUE},
+		{`"Hello".empty`, FALSE},
+		{`"Hello".eql("Hello")`, TRUE},
+		{`"Hello".eql("World")`, FALSE},
+		{`"Hello".start_with("Hel")`, TRUE},
+		{`"哈囉！世界！".start_with("哈囉！")`, TRUE},
+		{`"Hello".start_with("hel")`, FALSE},
+		{`"哈囉！世界".start_with("世界！")`, FALSE},
+		{`"Hello".end_with("llo")`, TRUE},
+		{`"哈囉！世界！".end_with("世界！")`, TRUE},
+		{`"Hello".end_with("ell")`, FALSE},
+		{`"哈囉！世界！".end_with("哈囉！")`, FALSE},
+		{`"Hello".insert(0, "X")`, "XHello"},
+		{`"Hello".insert(2, "X")`, "HeXllo"},
+		{`"Hello".insert(5, "X")`, "HelloX"},
+		{`"Hello".insert(-2, "X")`, "HelXlo"},
+		{`"Hello".insert(-6, "X")`, "XHello"},
+		{`"Hello".chop`, "Hell"},
+		{`"Hello\n".chop`, "Hello"},
+		{`"Hello".ljust(2)`, "Hello"},
+		{`"Hello".ljust(7)`, "Hello  "},
+		{`"Hello".ljust(10, "xo")`, "Helloxoxox"},
+		{`"Hello".rjust(2)`, "Hello"},
+		{`"Hello".rjust(7)`, "  Hello"},
+		{`"Hello".rjust(10, "xo")`, "xoxoxHello"},
+		{`"  Goby Lang   ".strip`, "Goby Lang"},
+		{`"\nGoby Lang\r\t".strip`, "Goby Lang"},
+		{`"Hello World".split("o")`, initArrayObject([]Object{initStringObject("Hell"), initStringObject(" W"), initStringObject("rld")})},
+		{`"Hello".split("")`, initArrayObject([]Object{initStringObject("H"), initStringObject("e"), initStringObject("l"), initStringObject("l"), initStringObject("o")})},
+		{`"Hello\nWorld\nGoby".split("\n")`, initArrayObject([]Object{initStringObject("Hello"), initStringObject("World"), initStringObject("Goby")})},
+		{`"Hello World".slice(1..6)`, "ello W"},
+		{`"1234567890".slice(6..1)`, ""},
+		{`"1234567890".slice(11..1)`, nil},
+		{`"1234567890".slice(11..-1)`, nil},
+		{`"1234567890".slice(-10..1)`, "12"},
+		{`"1234567890".slice(-5..1)`, ""},
+		{`"1234567890".slice(-10..-1)`, "1234567890"},
+		{`"1234567890".slice(-10..-11)`, ""},
+		{`"1234567890".slice(1..-1)`, "234567890"},
+		{`"1234567890".slice(1..-1234)`, ""},
+		{`"1234567890".slice(-11..5)`, nil},
+		{`"1234567890".slice(-11..-12)`, nil},
```

Comment:
@Maxwell-Alexius  The error happens in this test case, it seems like it'll still returns string object instead of `Null`